aws-sdk-java/LeoFSTest.java

```diff
@@ -0,0 +1,423 @@
+// Java Built-in
+import java.io.IOException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.BufferedInputStream;
+import java.io.RandomAccessFile;
+import java.security.MessageDigest;
+import java.util.List;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+// AWS-SDK-Java
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.Protocol;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.AccessControlList;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import com.amazonaws.services.s3.model.Bucket;
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.DeleteObjectsRequest;
+import com.amazonaws.services.s3.model.DeleteObjectsResult;
+import com.amazonaws.services.s3.model.DeleteObjectsResult.DeletedObject;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.Grant;
+import com.amazonaws.services.s3.model.ListObjectsRequest;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.amazonaws.services.s3.transfer.TransferManager;
+import com.amazonaws.services.s3.transfer.Upload;
+
+// Extra
+import org.apache.commons.io.IOUtils;
+
+public class LeoFSTest {
+    private static String host = "localhost";   // leo_gateway host
+    private static Integer port = 8080;         // leo_gateway port
+
+    private static String accessKeyId       = "05236";
+    private static String secretAccessKey   = "802562235";
+
+    private static String signVer   = "v4";
+
+    private static String bucket    = "testj";
+    private static String tempData  = "../temp_data/";
+
+    private static final String smallTestF  = tempData + "testFile";
+    private static final String largeTestF  = tempData + "testFile.large"; 
+
+    private static ClientConfiguration config;
+    private static AWSCredentials credentials;
+    private static AmazonS3 s3;
+
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+        // Init
```

Comment:
As we keep doing tests with authentication v2 for a while, 
Please make signVer configurable via a command line argument (not only java but all s3 clients except one which don't have v2 signer like aws-sdk-go.